torchvision/prototype/transforms/_augment.py

```diff
@@ -79,22 +78,22 @@ def _get_params(self, sample: Any) -> Dict[str, Any]:
 
             i = torch.randint(0, img_h - h + 1, size=(1,)).item()
             j = torch.randint(0, img_w - w + 1, size=(1,)).item()
+
+            needs_erase = True
+            erase_params = dict(i=i, j=j, h=h, w=w, v=v)
+
             break
         else:
-            i, j, h, w, v = 0, 0, img_h, img_w, image
```

Comment:
We return `image` here to replace the image with itself in the `erase` kernels. Instead of doing that, I've added the `needs_erase` parameter and don't call the dispatcher in case it is not set. This will also give us a performance boost while also remove the need to actually retrieve the image.

Reply:
Why can't we return `v = None` and thus do not erase ?

Reply:
You mean use `v is None` as check whether we want to erase or not instead of `needs_erase`? We used the `needs_x` idiom before

https://github.com/pytorch/vision/blob/a409c3fef429076168747d375614e10e72cf035f/torchvision/prototype/transforms/_geometry.py#L850

https://github.com/pytorch/vision/blob/a409c3fef429076168747d375614e10e72cf035f/torchvision/prototype/transforms/_geometry.py#L866

so I thought we can reuse it again to make it very clear what is happening. No strong opinion though.

Reply:
I do not think that this idiom is needed here, you define `erase_params = None` for non-erase case and `needs_erase` is useless variable, IMO.
On the other hand, params can remain `dict(i=i, j=j, h=h, w=w, v=v)` and thus to distinguish if we need erase or not, we can check for `v is None`

Reply:
Well, it is not needed in the cited cases either, but we went for it anyway. We could also use some sentinel values there as well. Let's have @datumbox break the tie :innocent:

Reply:
If I were to write this piece of code, I would also probably be passing `None`. But this approach is equally valid, so I'll leave it to the author to decide.

Reply:
I'll go with `None` then.